### ISSUE TITLE
Locks

### ISSUE THREAD

**Issue by bsideup** [Impetus/kundera#282]
Hello everyone!

Does Kundera supports Jpa locking? ( http://en.wikibooks.org/wiki/Java_Persistence/Locking#JPA_2.0_Locking )

Thanks!

**Comment by mevivs** [comment]
Nope. Do you any thoughts to share how actually such locking mechanism should work over nosql databases?

-Vivek

**Comment by bsideup** [comment]
http://wiki.apache.org/cassandra/Locking

Yep.

**Comment by mevivs** [comment]
It's bit old thread. Does Cassandra support this?

-Vivek

**Comment by bsideup** [comment]
Hard to say.

But AFAIK Hector  support this:
https://github.com/hector-client/hector/blob/master/core/src/test/java/me/prettyprint/cassandra/locking/HLockManagerImplTest.java

**Comment by mevivs** [comment]
AFAIK, Cassandra doesn't.

Such custom implementation at client level may be possible, AFN Kundera doesn't support this.

-Vivek

**Comment by OleksandrSemenov** [comment]
Hi

There are no locking in Cassandra, but if you want to implement it please take a look at the Cages library - https://code.google.com/p/cages/
Note, that it requires ZooKeeper

Regards,
Oleksandr

**Comment by pires** [comment]
Locking in a variable-consistency and multiple-client (each with a possible _lock_) doesn't seem a sane idea.

**Comment by asafd1** [comment]
actually that Cassandra now supports optimistic locking through lightweight-transactions - this could be implemented in Kundera. without such locking - how do you suggest to avoid persisting a stale entity?

**Comment by karthikprasad13** [comment]
Optimistic locking is one of the many advantages of lightweight transactions. IMHO handling it at higher level (modeling the databases accordingly) would be more appropriate than implementing it in a NoSql datastore mapper like Kundera.

This can be achieved using native querying on Cassandra through Kundera. Please let us know how important it is, your usecase and if we can discuss the best way to achieve it?

We can connect on [Gitter](https://gitter.im/impetus-opensource/Kundera)

-Karthik
